### PR TITLE
Add JournalPage tracking

### PR DESCRIPTION
**Pull request by tuchfarber**
Viewing a page in the journal will now POST to the UserPageVisit endpoint. This PR relies on edx/journals#97

### CODE REVIEW

File: src/components/JournalPage/index.jsx [hunk 1]
```diff
@@ -16,14 +16,25 @@ import PDFViewer from '../PDFViewer';
...
   componentDidMount() {
+    this.trackVisit();
     this.props.getPage(this.props.match.params.pageId);
   }
   componentDidUpdate(prevProps) {
+    if (prevProps.userId !== this.props.userId) {
+      this.trackVisit();
+    }
     if (prevProps.match.params.pageId !== this.props.match.params.pageId) {
+      this.trackVisit();
```
Comment (bill-filler) [inline src/components/JournalPage/index.jsx]: would we only want to track after we've confirmed the getPage() succeeded?

File: src/data/services/JournalsApiService.js [hunk 2]
```diff
@@ -30,6 +35,18 @@ class JournalsApiService {
...
   }
+
+  static setPageVisit(userId, pageId) {
+    return axios.post(`${JournalsApiService.apiUrl}/userpagevisits/`, {
+      user: userId,
+      page: pageId,
+    }, {
+      withCredentials: true,
+      headers: {
+        'X-CSRFToken': getCookieValue('journals_csrftoken'),
```
Comment (bill-filler) [inline src/data/services/JournalsApiService.js]: why do we need this for this api call only and not the others?

File: src/components/JournalPage/index.jsx (hunk 1 above)
Comment (bill-filler) [reply src/components/JournalPage/index.jsx]: i.e. on `fetchPageSuccess`

File: src/components/JournalPage/index.jsx [hunk 3]
```diff
@@ -16,14 +16,25 @@ import PDFViewer from '../PDFViewer';
 
 class JournalPage extends React.Component {
   componentDidMount() {
+    this.trackVisit();
```
Comment (bill-filler) [inline src/components/JournalPage/index.jsx]: same comment here about placement of trackVisit() call

File: src/components/JournalPage/index.jsx (hunk 1 above)
Comment (tuchfarber) [reply src/components/JournalPage/index.jsx]: That's a good point. Will fix.

File: src/data/services/JournalsApiService.js (hunk 2 above)
Comment (tuchfarber) [reply src/data/services/JournalsApiService.js]: Because POST calls are an "unsafe" call and only unsafe calls requires CSRF protection.

**Pull request merged**